azurerm/internal/services/eventhub/eventhub_namespace_dedicated_resource.go

```diff
@@ -0,0 +1,415 @@
+package eventhub
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/eventhub/mgmt/2018-01-01-preview/eventhub"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+// Default Authorization Rule/Policy created by Azure, used to populate the
+// default connection strings and keys
+var eventHubNamespaceDedicatedResourceName = "azurerm_eventhub_namespace_dedicated"
+
+func resourceArmEventHubNamespaceDedicated() *schema.Resource {
```

Comment:
after chatting internally - rather than splitting this out into it's own resource, since the only difference between these two resources is validation, could we add `dedicated_cluster_id` as an optional field to `azurerm_eventhub_namespace`

with regards to the validation differences - since we're already conditionally capturing this logic via the `create` call for network rules - we can catch this in the same manner for non-dedicated clusters during the `create` - so it should be possible to reuse these for the moment